runtime/astar/src/lib.rs

```diff
@@ -2041,13 +2041,18 @@ impl_runtime_apis! {
         fn trace_transaction(
             extrinsics: Vec<<Block as BlockT>::Extrinsic>,
             traced_transaction: &pallet_ethereum::Transaction,
-            _header: &<Block as BlockT>::Header,
+            header: &<Block as BlockT>::Header,
         ) -> Result<
             (),
             sp_runtime::DispatchError,
         > {
             use moonbeam_evm_tracer::tracer::EvmTracer;
 
+            // Initialize block: calls the "on_initialize" hook on every pallet
+            // in AllPalletsWithSystem is needed because pallet message queue was introduced,
+            // and it executes messages "on_initialize" instead of "setValidationData" it used to do before.
+            Executive::initialize_block(header);
+
```

Comment:
I thought since we don't handle EVM XCM case scenarios so we don't need this, did I miss something?

Reply:
we do, initiliaze needs to be called before applying tx. The comments are not well explained

Reply:
slot_number for example is set on initialize. I will update the comment.